addons/project_timesheet_holidays/models/hr_employee.py

```diff
@@ -19,15 +19,24 @@ def create(self, vals):
 
     def write(self, vals):
         result = super(Employee, self).write(vals)
-        if 'active' in vals:
-            if vals.get('active'):
-                # Create future holiday timesheets
-                self._create_future_public_holidays_timesheets(self)
-            else:
+        if 'active' in vals or 'resource_calendar_id' in vals:
+            has_calendar_future_holidays = 'resource_calendar_id' in vals and \
+                self.env['resource.calendar.leaves'].search_count([
+                    ('date_from', '>=', fields.date.today()),
+                    ('calendar_id', '=', vals['resource_calendar_id'])
+                ])
+            if not vals.get('active') or has_calendar_future_holidays:
```

Comment:
```suggestion
            if not vals['active'] or has_calendar_future_holidays:
```
little detail

Reply:
No this we gives a error when `active` is not in vals(when we change `resource_calendar_id`)

Reply:
my mistake :+1: